R/decisiontreeutils.R

```diff
@@ -0,0 +1,431 @@
+  prune_tbl1 <- out[strt_table:length(out)]
+  out <- c(the_call, out[(end_call + 1):(strt_table - 2)])
+  rpart_out <- data.frame(grp = c("Call", rep("Model_Sum", model_sum)), out = out)
+  rpart_out$grp <- as.character(rpart_out$grp)
+  rpart_out$out <- as.character(out)
+  # Pipe delimit the pruning table and then rbind it to the output
+  prune_tbl <- NULL
+  for (i in 1:length(prune_tbl1)) {
+    a_row <- unlist(strsplit(prune_tbl1[i], "\\s"))
+    a_row <- a_row[a_row != ""]
+    prune_tbl <- c(prune_tbl, paste(a_row[1], a_row[2], a_row[3], a_row[4], a_row[5], a_row[6], sep="|"))
+  }
+  pt_df <- data.frame(grp = rep("Prune", length(prune_tbl)), out = prune_tbl)
+  pt_df$grp <- as.character(pt_df$grp)
+  pt_df$out <- as.character(pt_df$out)
+  rpart_out <- rbind(rpart_out, pt_df)
+
+  # The leaf summary
+  if (is_XDF) {
+    print(the_model_rpart) # Tree Leaf Summary
+    leaves <- capture.output(the_model_rpart)
+  } else {
+    print(the_model) # Tree Leaf Summary
+    leaves <- capture.output(the_model)
+  }
+  leaves_num <- 1:length(leaves)
+  start_leaves <- leaves_num[substr(leaves, 1, 5) == "node)"]
+  leaves <- leaves[start_leaves:length(leaves)]
+  leaves <- gsub(">", "&gt;", leaves)
+  leaves <- gsub("<", "&lt;", leaves)
+  leaves <- gsub("\\s", "<nbsp/>", leaves) # Order matters
+  leaves_df <- data.frame(grp = rep("Leaves", length(leaves)), out = leaves)
+  leaves_df$grp <- as.character(leaves_df$grp)
+  leaves_df$out <- as.character(leaves_df$out)
+
+  rpart_out <- rbind(rpart_out, leaves_df)
+
+  # Indicate that this is an object of class rpart or rxDTree
+  if (is_XDF) {
+    rpart_out <- rbind(c("Model_Name", config$model.name), rpart_out, c("Model_Class", "rxDTree"))
+  } else {
+    rpart_out <- rbind(c("Model_Name", config$model.name), rpart_out, c("Model_Class", "rpart"))
+  }
+
+  # Write out the grp-out table for reporting
+  # results$out1 <- rpart_out
+  rpart_out
+}
+
+#' get graphing calls
+#'
+#' @param config list of config options
+#' @param the_model model object
+#' @param is_XDF boolean of whether model is XDF
+#' @return params to call AlteryxGraph on
+getDTGraphCalls <- function(config, the_model, is_XDF) {
+  # Address the user plot parameters and create the plots
+
+  # The values in the leaf summary
+  leaf_sum <- 4
+  if (config$do.counts == "True")
+    leaf_sum <- 2
+  if (the_model$method != "class")
+    leaf_sum <- 0
+  print(the_model$method)
+
+  # Uniform or proportional tree branch lengths
+  uniform <- "FALSE"
+  fallen <- "TRUE"
+  if (config$b.dist == "True") {
+    uniform <- "TRUE"
+    fallen <- "FALSE"
+  }
+
+  # assemble list of function and params to pass to AlteryxGraph
+  graph_calls <- list()
+
+  graph_calls$tree <- list()
+  graph_calls$tree$f <- "rpart.plot"
+  graph_calls$tree$args <- list (
+    the_model_rpart,
+    type = 0,
+    extra = leaf_sum,
+    uniform = uniform,
+    fallen.leaves = fallen,
+    main = "Tree Plot",
+    cex = 1
+  )
+
```

Comment:
A cleaner solution that would be testable is to make the graphing calls functions. For example

```r
graph_calls$tree <- function(){
  rpart.plot(the_model_rpart, type = 0, extra = leaf_sum, uniform = uniform,
    fallen.leaves = fallen, main = "Tree Plot", cex = 1
  )
}
```

You  can now call it as `graph_calls$tree()` inside `AlteryxGraph` or probably `AlteryxGraph2`.